platform/core/src/services/ToolBarService/ToolbarService.ts

```diff
@@ -2,35 +2,43 @@ import merge from 'lodash.merge';
 import { CommandsManager } from '../../classes';
 import { ExtensionManager } from '../../extensions';
 import { PubSubService } from '../_shared/pubSubServiceInterface';
-import type { RunCommand, Commands } from '../../types/Command';
+import type { RunCommand, Commands, Command } from '../../types/Command';
 
 const EVENTS = {
   TOOL_BAR_MODIFIED: 'event::toolBarService:toolBarModified',
   TOOL_BAR_STATE_MODIFIED: 'event::toolBarService:toolBarStateModified',
 };
 
-export type ButtonListeners = Record<string, RunCommand>;
-
-export interface ButtonProps {
-  primary?: Button;
-  secondary?: Button;
-  items?: Button[];
+enum ButtonInteractionType {
+  ACTION = 'action',
+  TOGGLE = 'toggle',
+  TOOL = 'tool',
 }
 
-export interface Button extends Commands {
+export interface ButtonProps {
   id: string;
-  icon?: string;
-  label?: string;
-  type?: string;
+  icon: string;
+  label: string;
   tooltip?: string;
+  commands?: Command | Commands;
+  type?: ButtonInteractionType;
   isActive?: boolean;
-  listeners?: ButtonListeners;
-  props?: ButtonProps;
+  listeners?: Record<string, RunCommand>;
 }
 
-export interface ExtraButtonOptions {
-  listeners?: ButtonListeners;
-  isActive?: boolean;
+export interface NestedButtonProps {
+  groupId: string;
+  items: ButtonProps[];
+  primary: ButtonProps;
+  secondary: ButtonProps;
+}
+
+export interface Button {
+  id: string;
+  props: ButtonProps | NestedButtonProps;
```

Comment:
If this is going to change so much, can we make the button just contain it's props directly, and use two type includes?